File: android/ValleTPV/app/src/main/java/com/valleapp/valletpv/adaptadoresDatos/AdaptadorSelCam.java
package com.valleapp.valletpv.adaptadoresDatos;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.TextView;

import com.valleapp.valletpv.R;

import org.json.JSONException;
import org.json.JSONObject;

import java.util.List;

/**
 * Created by valle on 14/09/14.
 */
public class AdaptadorSelCam extends ArrayAdapter<JSONObject> {

    Context cx;
    List<JSONObject> values;

    public AdaptadorSelCam(Context context, List<JSONObject> obj) {
        super(context, R.layout.item_simple, obj);
        this.cx = context;
        this.values= obj;
    }

    @Override
    public View getView(int position, View convertView, ViewGroup parent) {

        LayoutInflater inflater = (LayoutInflater) cx
                .getSystemService(Context.LAYOUT_INFLATER_SERVICE);
        View rowView = inflater.inflate(R.layout.item_simple, parent, false);

        try {
            JSONObject obj = values.get(position);
            TextView nombre = (TextView) rowView.findViewById(R.id.texto_linea);
            nombre.setText(String.format("%s",obj.getString("Nombre")));
            rowView.setTag(obj);

        } catch (JSONException e) {
            e.printStackTrace();
        }


        return rowView;
    }
}

File: android/ValleTPV/ValleTPV/src/main/java/com/valleapp/valletpv/adaptadoresDatos/AdaptadorSelCam.kt
package com.valleapp.valletpv.adaptadoresDatos

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ArrayAdapter
import android.widget.TextView
import com.valleapp.valletpv.R
import org.json.JSONObject

class AdaptadorSelCam(
    context: Context,
    private val values: List<JSONObject>
) : ArrayAdapter<JSONObject>(context, R.layout.item_simple, values) {

    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
        val rowView = convertView ?: LayoutInflater.from(context).inflate(R.layout.item_simple, parent, false)

        try {
            val obj = values[position]
            val nombre = rowView.findViewById<TextView>(R.id.texto_linea)
            nombre.text = "${obj.getString("nombre")} ${obj.getString("apellidos")}"
            rowView.tag = obj // Guardamos el JSONObject directamente en el tag
        } catch (e: Exception) {
            e.printStackTrace()
        }

        return rowView
    }
}
